toast-tk-rest-utils/src/main/java/io/toast/tk/core/rest/HttpRequest.java

```diff
@@ -0,0 +1,97 @@
+package io.toast.tk.core.rest;
+
+public class HttpRequest {
+
+	private final String uri;
+	private String apiKey;
+	private String json;
+	private String proxyAddress;
+	private int proxyPort;
+	private String proxyUser;
+	private String proxyPassword;
+
+	public static class Builder {
+		private String uri;
+		private String apiKey;
+		private String json;
+
+		private Builder(){
+		}
+
+		public static Builder create(){
+			return new Builder();
+		}
+
+		public Builder uri(String uri){
+			this.uri = uri;
+			return this;
+		}
+
+		public Builder json(String json){
+			this.json = json;
+			return this;
+		}
+
+		public Builder withKey(String apiKey){
+			this.apiKey = apiKey;
+			return this;
+		}
+
+		public HttpRequest build(){
+			HttpRequest request = new HttpRequest(uri, json);
+			request.setApiKey(apiKey);
+			return request;
+		}
+	}
+
+	public HttpRequest(String uri) {
+		this.uri = uri;
+	}
+	
+	public HttpRequest(String uri, String json) {
+		this(uri);
+		this.json = json;
+	}
+
+	public void setProxyInfo(String proxyHost, 
+			int proxyPort2, 
+			String proxyUser2, 
+			String proxyPassword2) {
+		this.proxyAddress = proxyHost;
+		this.proxyPort = proxyPort2;
+		this.proxyUser = proxyUser2;
+		this.proxyPassword = proxyPassword2;
+	}
+
+	private void setApiKey(String apiKey) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this method into "Builder". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3398)